apps/backend/backend-main/src/entity/PromotionCode.ts

```diff
@@ -0,0 +1,45 @@
+import { Column, Entity, ManyToOne, Unique } from "typeorm";
+
+import { OrderTypeEnum } from "../shared/enums";
+import { BaseEntity } from "./BaseEntity";
+import { User } from "./User";
+import { Max, Min } from "class-validator";
+
+@Entity()
+@Unique(["code"])
+export class PromotionCode extends BaseEntity {
+  @Column({ type: "varchar", length: 20 })
+  code: string;
+
+  @Column({ type: "varchar", nullable: true })
+  description: string;
+
+  @Min(0)
+  @Column({ type: "float", nullable: true })
```

Comment:
Let's keep this nullable true. Because a percentage may or may not be the case. However, Amount is something that will always be there, either directly or calculated from the percentage

```suggestion
  @Column({ type: "float", nullable: false })
```